contracts/vaults/UniV3Vault.sol

```diff
@@ -183,11 +184,16 @@ contract UniV3Vault is IUniV3Vault, IntegrationVault {
     {
         address[] memory tokens = _vaultTokens;
         for (uint256 i = 0; i < tokens.length; ++i) {
-            _allowTokenIfNecessary(tokens[i], address(_positionManager));
+            IERC20(tokens[i]).safeIncreaseAllowance(address(_positionManager), tokenAmounts[i]);
         }
 
         actualTokenAmounts = new uint256[](2);
-        if (uniV3Nft == 0) return actualTokenAmounts;
+        if (uniV3Nft == 0) {
```

Comment:
Can we move that condition before safeIncreaseAllowance and avoid safeApprove(0)?